metapackages/auto-instrumentations-node/src/utils.ts

```diff
@@ -136,3 +161,63 @@ export function getNodeAutoInstrumentations(
 
   return instrumentations;
 }
+
+export function getResourceDetectorsFromEnv(): Array<Detector | DetectorSync> {
+  const resourceDetectorsFromEnv =
+    process.env.OTEL_NODE_RESOURCE_DETECTORS?.split(',');
+
+  if (
+    resourceDetectorsFromEnv == null ||
+    resourceDetectorsFromEnv.includes('all')
+  ) {
+    return [
+      // Standard resource detectors.
+      containerDetector,
+      envDetectorSync,
+      hostDetectorSync,
+      osDetectorSync,
+      processDetectorSync,
+      // Cloud resource detectors.
+      alibabaCloudEcsDetector,
+      // Ordered AWS Resource Detectors as per:
```

Comment:
I think it node, in contast to the collector, the last detector to set an attribute wins:

Collector: 
> Note that if multiple detectors are inserting the same attribute name, the first detector to insert wins

Node:
Using the merge function which:
> If a key exists on both the old and updating resource, the value of the updating resource MUST be picked

Thus I think it should be reversed, e.g. first EC2 and the EKS. Did you check it by any chance?